File: backend/app/schemas/creator.py
from typing import Optional
from pydantic import BaseModel


class PublicAddonResponse(BaseModel):
    key: str
    label: str
    price: int

    model_config = {"from_attributes": True}


class PublicPackageResponse(BaseModel):
    id: str
    tier: str
    name: str
    price: int
    deliverables: list[str]
    revisions: int
    delivery_days: int
    usage_rights: str
    analytics: bool

    model_config = {"from_attributes": True}


class PublicPortfolioResponse(BaseModel):
    id: str
    title: str
    kind: str
    url: str
    thumbnail_url: Optional[str] = None
    meta: Optional[str] = None
    sort_order: int = 0

    model_config = {"from_attributes": True}


class PublicCreatorResponse(BaseModel):
    handle: str
    display_name: Optional[str] = None
    avatar_url: Optional[str] = None
    bio: Optional[str] = None
    cover_url: Optional[str] = None
    location: Optional[str] = None
    languages: list[str] = []
    genres: list[str] = []
    platforms: dict[str, str] = {}
    total_reach: Optional[int] = None
    avg_engagement: Optional[float] = None
    on_time_rate: Optional[float] = None
    brands_count: int = 0
    packages: list[PublicPackageResponse] = []
    addons: list[PublicAddonResponse] = []
    portfolio: list[PublicPortfolioResponse] = []

File: frontend/types/creator.ts
export interface PublicPackage {
  id: string;
  tier: "basic" | "standard" | "premium" | "campaign";
  name: string;
  price: number; // paise
  deliverables: string[];
  revisions: number;
  delivery_days: number;
  usage_rights: "personal" | "commercial";
  analytics: boolean;
}

export interface PublicAddon {
  key: string;
  label: string;
  price: number; // paise
}

export interface PublicPortfolioItem {
  id: string;
  title: string;
  kind: "image" | "video" | "link" | "pdf";
  url: string;
  thumbnail_url?: string;
  meta?: string;
  sort_order: number;
}

export interface PublicCreator {
  handle: string;
  display_name?: string;
  avatar_url?: string;
  bio?: string;
  cover_url?: string;
  location?: string;
  languages: string[];
  genres: string[];
  platforms: Record<string, string>;
  total_reach?: number;
  avg_engagement?: number;
  on_time_rate?: number;
  brands_count: number;
  packages: PublicPackage[];
  addons: PublicAddon[];
  portfolio: PublicPortfolioItem[];
}
